src/boot/bootctl.c

```diff
@@ -1708,7 +1708,11 @@ static int verb_set_default(int argc, char *argv[], void *userdata) {
 static int verb_random_seed(int argc, char *argv[], void *userdata) {
         int r;
 
-        r = acquire_esp(false, NULL, NULL, NULL, NULL);
+        r = find_esp_and_warn(arg_esp_path, false, &arg_esp_path, NULL, NULL, NULL, NULL);
+        if (r == -ENOKEY) {
+                log_notice("No ESP found, not initializing random seed.");
+                return 0;
+        }
```

Comment:
The conditionalization is really hard, since the ESP might be mounted at a variety of places really. we actively search for it. Note that you will still see a log msg if we skip operation, so it's not that it's entirely silent and invisible...

Reply:
Then I think we need to check if we are running as part of the service, and only change behaviour in that case. I think making this explicit would be best: add a command line option `--skip-if-no-efi`, and call `bootctl random-seed --skip-if-no-efi` in the unit.